GVFS/GVFS.Common/Git/GitProcess.cs

```diff
@@ -108,6 +109,23 @@ public static bool TryGetVersion(string gitBinPath, out GitVersion gitVersion, o
             return true;
         }
 
+        public bool KillRunningProcess()
+        {
+            try
+            {
+                if (this.executingProcess != null)
+                {
+                    this.executingProcess.Kill();
+                    return true;
+                }
+            }
+            catch (NullReferenceException)

```

Comment:
Same here, you should be able to drop this if you use an interlocked compare/exchange to set the `executingProcess` to null.